quicklogic/qlf_k4n8/devices/umc22/qlf_k4n8_umc22-arch.xml

```diff
@@ -0,0 +1,7998 @@
+<!-- 
```

Comment:
Was this file generated by hand?  Seems like portions of this should be refactored into some generator scripts.  Thoughts?

Reply:
Yes, this architecture has been written manually and then processed by a script. The flow of  processing the architecture is still under development. I think that subsequent versions may have all comments stripped.